From 3aa447ae8b331acfa307f8f5dfbee2b2be66f6b3 Mon Sep 17 00:00:00 2001
From: Karthikeyan Kathirvel <quic_kathirve@quicinc.com>
Date: Mon, 1 Apr 2024 15:55:50 +0530
Subject: [PATCH] hostapd: Update DTIM information properly in per-STA profile

The DTIM information in the per-STA profile is being incorrectly
updated. The DTIM period is being updated in the LSB byte of the
DTIM information (2 bytes), which is intended for the DTIM count.

Fix by updating the DTIM period and DTIM count information
properly to the MSB and LSB bytes of the DTIM information
respectively.

Signed-off-by: Karthikeyan Kathirvel <quic_kathirve@quicinc.com>
---
 src/ap/ieee802_11_eht.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/ap/ieee802_11_eht.c b/src/ap/ieee802_11_eht.c
index bb02b4b..3121bc7 100644
--- a/src/ap/ieee802_11_eht.c
+++ b/src/ap/ieee802_11_eht.c
@@ -766,7 +766,8 @@ u8 * hostapd_eid_eht_basic_ml(struct hostapd_data *hapd, u8 *eid,
 		wpabuf_put_le64(buf, 0);
 
 		/* DTIM Info */
-		wpabuf_put_le16(buf, link_bss->conf->dtim_period);
+		wpabuf_put_u8(buf, 0); /* DTIM Count */
+		wpabuf_put_u8(buf, link_bss->conf->dtim_period);
 
 		/* BSS Parameters Change Count */
 		if (type == WLAN_FC_STYPE_ASSOC_RESP || type == WLAN_FC_STYPE_REASSOC_RESP)
-- 
2.34.1

